src/components/navbar/navbar.component.tsx

```diff
@@ -55,6 +57,18 @@ const Navbar: React.FC<NavbarProps> = ({
     }
   };
 
+  const [switcherToggler, setSwitcherToggler] = React.useState(false);
+
+  const handleSwitcherClick = React.useCallback(() => {
+    setSwitcherToggler(switcherToggler => !switcherToggler);
+  }, []);
+
+  const icon = switcherToggler ? (
```

Comment:
Could be shorter if you just follow my previous comment.

```
const Icon = switchToggler ? Close20 : AppSwitcher20;

// later
<Icon onClick={handleSwitcherClick} />
```

Reply:
Sir, I found difficulties implementing it like this. I got some typescript errors. I tried figuring out how to do this. But couldn't implement it exactly like this.

Reply:
It works. Just tell TS that this is a `React.FC<any>` or specify the type of the props (but I guess its anyway `any` from the beginning on).

Also make sure to have it named like I have. Know your JSX rule (e.g., `Icon` not `icon`).

Reply:
I have Implemented it Sir. Thank you very much for guiding me along. I have learnt Alot.